topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -406,8 +413,56 @@ private TopicStore createTopicStore(Zk zk, Config config) {
     }
 
     private void setSaslConfigs(Properties kafkaClientProps) {
-        String saslMechanism;
-        String jaasConfig;
+        String configSaslMechanism = config.get(Config.SASL_MECHANISM);
+        
+        switch (configSaslMechanism) {
+            case SASL_TYPE_PLAIN:
+            case SASL_TYPE_SCRAM_SHA_256:
+            case SASL_TYPE_SCRAM_SHA_512:
+                setStandardSaslConfigs(kafkaClientProps);
+                break;
+            case SASL_TYPE_CUSTOM:
+                setCustomSaslConfigs(kafkaClientProps);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid SASL_MECHANISM: '" + configSaslMechanism + "'");
+        }
+    }
+
+    private void setCustomSaslConfigs(Properties kafkaClientProps) {
+        String customPropsString = config.get(Config.SASL_CUSTOM_CONFIG);
+
+        if (customPropsString.isEmpty()) {
+            throw new InvalidConfigurationException(Config.SASL_MECHANISM.key + "=custom, but env var " + Config.SASL_CUSTOM_CONFIG.key + " is not set");
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Map<String, String> customProperties = objectMapper.readValue(customPropsString, STRING_HASH_MAP_TYPE_REFERENCE);
+
+            if (customProperties.isEmpty()) {
+                throw new InvalidConfigurationException(Config.SASL_MECHANISM.key + "=custom, but env var " + Config.SASL_CUSTOM_CONFIG.key + " is empty");
+            }
+
+            for (var entry : customProperties.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+
+                if (key == null || key.isBlank() || !key.startsWith("sasl.")) {
+                    throw new InvalidConfigurationException("SASL custom config property '" + key + "' in env var " + Config.SASL_CUSTOM_CONFIG.key + " does not start with 'sasl.'");
+                }
+
+                kafkaClientProps.setProperty(key, value);
+            }
+        } catch (JsonProcessingException e) {
+            var message = e.getMessage();
+            throw new InvalidConfigurationException("Env var " + Config.SASL_CUSTOM_CONFIG.key + " could not be parsed as JSON" + (message != null ? ": " + e.getMessage() : "."));
+        }
+    }
+
+    private void setStandardSaslConfigs(Properties kafkaClientProps) {
```

Comment:
Can we validate in this method that `SASL_CUSTOM_CONFIG` is _not_ set, with a nice error message that it's only available with the `custom` mechanism?

Similar in the `v2` package.